### PR TITLE
fix: use podman in rootful mode on linux

### LINKED ISSUE #2715: Make sure Podman installation works on Linux and Windows
The Podman dependency needs to be thoroughly tested on all OS and any issues have to be resolved.

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

This PR fixes the `solo one-shot deploy` flow when using Podman on Linux and Windows.

When the host machine does not have Docker installed, Podman is installed as a package using `brew` and is used in rootfull mode to create a cluster with `kind`.

Adds a new github workflow that tests all automatic external dependency installations - `kubectl`, `helm`, `kind`, `podman` using a default Ubuntu v24 runner.

The PR also adds a new flag to `one-shot single deploy`: `--minimal-setup`. When set to true, solo will create a deployment without `Explorer` and `JSON-RPC relay`. This saves on resources and ensures that a default GitHub runner can handle a one-shot deployment.

### Related Issues

* Closes #2715 
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

**Review by instamenta — APPROVED** [review]

File: src/core/dependency-injection/inject-tokens.ts [hunk 1]
```diff
@@ -83,6 +83,9 @@ export class InjectTokens {
   public static Commands: symbol = Symbol.for('Commands');
   public static MetricsServer: symbol = Symbol.for('MetricsServer');
   public static BackupRestoreCommand: symbol = Symbol.for('BackupRestoreCommand');
+  public static OSPackageManager: symbol = Symbol.for('OSPackageManager');
```
Comment (jeromy-cannon) [inline src/core/dependency-injection/inject-tokens.ts]:
```suggestion
  public static OsPackageManager: symbol = Symbol.for('OsPackageManager');
```

**Review by jeromy-cannon — CHANGES_REQUESTED** [review]

**Review by jeromy-cannon — APPROVED** [review]
